lib/phoenix_live_view/async.ex

```diff
@@ -180,6 +180,74 @@ defmodule Phoenix.LiveView.Async do
     |> run_async_task(keys, wrapped_func, :assign, opts)
   end
 
+  def stream_async(socket, key, func, opts, env) do
+    validate_function_env(func, :stream_async, env)
+
+    quote do
+      Phoenix.LiveView.Async.stream_async(
+        unquote(socket),
+        unquote(key),
+        unquote(func),
+        unquote(opts)
+      )
+    end
+  end
+
+  def stream_async(%Socket{} = socket, key, func, opts)
+      when is_atom(key) and is_function(func, 0) do
+    # runtime check
+    if Phoenix.LiveView.connected?(socket) do
+      validate_function_env(func, :stream_async)
+    end
+
+    reset = Keyword.get(opts, :reset, false)
+
+    check_enumerable! = fn result ->
+      if Enumerable.impl_for(result) do
+        :ok
+      else
+        raise ArgumentError, """
+        expected stream_async to return {:ok, Enumerable.t()} or {:error, reason} but the result
+        is does not implement the Enumerable protocol
+        """
+      end
+    end
+
+    # store opts for streaming
+    wrapped_func = fn ->
+      case func.() do
+        {:ok, result} ->
+          check_enumerable!.(result)
+          {:ok, result, []}
+
+        {:ok, result, opts} when is_list(opts) ->
+          check_enumerable!.(result)
+          {:ok, result, opts}
+
+        {:error, reason} ->
+          {:error, reason}
+
+        other ->
+          raise ArgumentError, """
+          expected stream_async to return {:ok, Enumerable.t()} or {:error, reason}, got: #{inspect(other)}
+          """
+      end
+    end
+
+    value =
+      case {reset, socket.assigns} do
+        {false, %{^key => %AsyncResult{ok?: true} = existing}} ->
+          AsyncResult.loading(existing)
+
+        _ ->
+          AsyncResult.loading()
+      end
+
+    socket
+    |> Phoenix.Component.assign(key, value)
```

Comment:
I may be missing it... but shouldn't you call `stream_configure` and define the stream upfront as well? The stream should be defined since the beginning IMO...